google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1705,148 +1710,12 @@ private Value encodeValue(FieldPath fieldPath, Object value) {
-              lastReceivedDocument.set(documentSnapshot);
-            }
-
-            if (response.getDone()) {
-              currentSpan.addEvent(
-                  TraceUtil.SPAN_NAME_RUN_QUERY + ": Received RunQueryResponse.Done");
-              onComplete();
-            }
-          }
-
-          @Override
-          public void onError(Throwable throwable) {
-            QueryDocumentSnapshot cursor = lastReceivedDocument.get();
-            if (shouldRetry(cursor, throwable)) {
-              currentSpan.addEvent(
-                  TraceUtil.SPAN_NAME_RUN_QUERY + ": Retryable Error",
-                  Collections.singletonMap("error.message", throwable.getMessage()));
-
-              Query.this
-                  .startAfter(cursor)
-                  .internalStream(
-                      runQueryResponseObserver,
-                      startTimeNanos,
-                      /* transactionId= */ null,
-                      options.getRequireConsistency() ? cursor.getReadTime() : null,
-                      explainOptions,
-                      /* isRetryRequestWithCursor= */ true);
-            } else {
-              currentSpan.addEvent(
-                  TraceUtil.SPAN_NAME_RUN_QUERY + ": Error",
-                  Collections.singletonMap("error.message", throwable.getMessage()));
-              runQueryResponseObserver.onError(throwable);
-            }
-          }
-
-          @Override
-          public void onComplete() {
-            if (hasCompleted) return;
-            hasCompleted = true;
-            currentSpan.addEvent(
-                TraceUtil.SPAN_NAME_RUN_QUERY + ": Completed",
-                Collections.singletonMap(ATTRIBUTE_KEY_DOC_COUNT, numDocuments));
-            runQueryResponseObserver.onCompleted();
-          }
-
-          boolean shouldRetry(DocumentSnapshot lastDocument, Throwable t) {
-            if (lastDocument == null) {
-              // Only retry if we have received a single result. Retries for RPCs with initial
-              // failure are handled by Google Gax, which also implements backoff.
-              return false;
-            }
-
-            // Do not retry EXPLAIN requests because it'd be executing
-            // multiple queries. This means stats would have to be aggregated,
-            // and that may not even make sense for many statistics.
-            if (explainOptions != null) {
-              return false;
-            }
-
-            Set<StatusCode.Code> retryableCodes =
-                FirestoreSettings.newBuilder().runQuerySettings().getRetryableCodes();
-            return shouldRetryQuery(t, transactionId, startTimeNanos, retryableCodes);
-          }
-        };
-
-    rpcContext.streamRequest(request.build(), observer, rpcContext.getClient().runQueryCallable());
-  }
-
   /**
    * Executes the query and returns the results as QuerySnapshot.
    *
    * @return An ApiFuture that will be resolved with the results of the Query.
    */
-  @Nonnull
```

Comment:
I think we still want this to be `@Nonnull` ?

Reply:
good catch. Super annoying that you can change this attribute in an override